tests/test_neural_linear.py

```
@@ -0,0 +1,328 @@
+def test_backbone_heads_meta_model_update_changes_state(
+    meta_model: BackboneHeadsMetaModel, rng: np.random.Generator
+) -> None:
+    """After ``update`` the backbone weights and at least one head posterior mean change."""
+    weights_before = [w.copy() for w in meta_model.backbone._params["W"]]
+    mu_before = {a: meta_model.actions[a].mu.copy() for a in meta_model.action_ids}
+
+    actions, rewards, context = _balanced_batch(rng)
+    meta_model.update(actions=actions, rewards=rewards, context=context)
+
+    backbone_changed = any(
+        not np.allclose(before, after) for before, after in zip(weights_before, meta_model.backbone._params["W"])
```

Ruff (0.15.17)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)